### PR TITLE
performance: prepare site for SSR

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] (if relevant) I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] (if relevant) I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3997** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3997" target="_blank">Graphite</a> 
    * **PR #3998** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3998" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3998?utm_source=stack-comment).

File: src/contexts/Announcements/AnnouncementLauncher.tsx [hunk 1]
```diff
@@ -35,8 +39,10 @@ export const AnnouncementLauncher: React.FC<
 
   // Try activating any announcements
   useEffect(() => {
-    // Activate first announcement that fits conditions
-    setActiveId(Announcements.find(shouldActivateAnnouncement)?.id)
+    startTransition(() => {
```
Comment (peripheralist) [inline src/contexts/Announcements/AnnouncementLauncher.tsx:42]: niiice

File: src/components/Navbar/SiteNavigation.tsx [hunk 2]
```diff
@@ -211,7 +211,7 @@ const ResourcesMenu = () => {
     <DropdownMenu
       className="text-base md:text-sm"
       items={resourcesMenuItems}
-      heading={t`Resources`}
+      heading={<Trans>Resources</Trans>}
```
Comment (peripheralist) [inline src/components/Navbar/SiteNavigation.tsx:214]: is there a reason for this related to SSR?

File: src/contexts/Theme/useJuiceTheme.ts [hunk 3]
```diff
@@ -21,15 +21,23 @@ const getInitialThemeOption = (storageKey: string) => {
...
+    'light' as ThemeOption,
+  )
 
   const isMobile = useMedia('(max-width: 767px)')
 
+  // Load the theme from local storage on initial load
+  useEffect(() => {
+    const initialThemeOption = getInitialThemeOption(storageKey)
+    startTransition(() => {
+      setCurrentThemeOption(initialThemeOption)
```
Comment (peripheralist) [inline src/contexts/Theme/useJuiceTheme.ts:34]: ⚡

**Review by peripheralist — APPROVED** [review]

**Pull request merged**